solarwinds_apm/oboe/configuration.py

```diff
@@ -0,0 +1,152 @@
+# © 2025 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+from __future__ import annotations
+
+from collections.abc import Callable
+
+
+class TransactionSetting:
+    def __init__(self, tracing: bool, matcher: Callable[[str], bool]):
+        self._tracing = tracing
+        self._matcher = matcher
+
+    @property
+    def tracing(self) -> bool:
+        return self._tracing
+
+    @tracing.setter
+    def tracing(self, value: bool):
+        self._tracing = value
+
+    @property
+    def matcher(self) -> Callable[[str], bool]:
+        return self._matcher
+
+    @matcher.setter
+    def matcher(self, value: Callable[[str], bool]):
+        self._matcher = value
+
+    def __str__(self):
+        return f"TransactionSetting(tracing={self._tracing}, matcher={self._matcher})"
+
+
+class Configuration:
+    def __init__(
+        self,
+        enabled: bool,
+        service: str,
+        token: str | None,
+        collector: str,
+        headers: dict[str, str],
+        log_level: int,
+        tracing_mode: bool | None,
+        trigger_trace_enabled: bool,
+        export_logs_enabled: bool,
+        transaction_name: Callable[[], str] | None,
+        transaction_settings: list[TransactionSetting],
+    ):
+        self._enabled = enabled
+        self._service = service
+        self._token = token
```

Comment:
To confirm: it will be the caller's responsibility to provide `service` and `token`, from whatever source (service key, Otel resource, etc)?

Reply:
After taking a deeper look, I think we don't need `token`. Removed 
HttpSampler required the service (i.e. the name) and the headers `{ "Authorization": f"Bearer {bearer}"`, where the token == bearer. 
JsonSampler required the settings file only